chunk/chunk_store.go

```diff
@@ -190,15 +197,125 @@ func hashValue(userID, bucket string, metricName model.LabelValue) string {
 	return fmt.Sprintf("%s:%s:%s", userID, bucket, metricName)
 }
 
-func rangeValue(label model.LabelName, value model.LabelValue, chunkID string) ([]byte, error) {
-	return lex.Encode(string(label), string(value), chunkID)
+func rangeValue(label model.LabelName, value model.LabelValue, chunkID string) []byte {
+	// encoded length is len(label) + 1 + len(encoded(value)) + 1 + len(chunkID) + 1 + version byte + 1
```

Comment:
Took me a while to parse this comment. I think it'd be more helpful to put a comment above the `length =` line explaining where `5` comes from. e.g.

```golang
// 5 = 4 field terminators + 1 version bytes
```

Or whatever.